src/components/document-id/DocumentId.vue

```diff
@@ -0,0 +1,256 @@
+    if (currentAccount) {
+      const accountInfo = JSON.parse(currentAccount)
+      config.headers['Account-Id'] = accountInfo.id
+    }
+
+    try {
+      // Make the API call to validate the Document ID. If the call is successful(200), the Document ID is not unique
+      const test = await axios.get(url, config)
+      return test.status
+    } catch (error: any) {
+      // If the API call fails, return the status code
+      if ([400, 404].includes(error.response.status)) {
+        return error.response.status
+      }
+    }
+  }
+
+  /**
+   * Validates the Document ID if its length is 8 characters.
+   * Depending on the API response, it sets the appropriate error messages and verification status.
+   * If the Document ID is verified, it emits the Document ID.
+   */
+  @Watch('documentId')
+  async onDocumentIdComplete (): Promise<void> {
+    // Validate the document id if the length is 8 characters
+    if (this.documentId?.length === 8) {
+      this.isVerifyingDocId = true
+      // Validate the document ID
+      const documentIdStatus = await this.fetchDocumentIdStatus()
+      switch (documentIdStatus) {
+        case 200:
+          this.docIdError = ['Must be unique number']
+          this.isVerifiedDocId = false
+          break
+        case 400:
+          this.docIdError = ['Document ID is invalid']
+          this.isVerifiedDocId = false
+          break
+        case 404:
+          this.docIdError = null
+          this.isVerifiedDocId = true
+          break
+        default:
+          this.docIdError = ['Document ID request failed']
+          this.isVerifiedDocId = false
+      }
+
+      // Emit the document ID if it is verified
+      !!this.isVerifiedDocId && this.emitDocumentId(this.documentId)
+    } else {
+      // Reset verification status and error display if the document ID length is not 8
+      this.isVerifiedDocId = false
+      this.emitDocumentId('')
+    }
+
+    // Reset the verifying status
+    this.isVerifyingDocId = false
+  }
+
+  /** Resets the documentId, isVerifyingDocId, and docIdError properties when `generateDocumentId` changes. */
+  @Watch('generateDocumentId')
+  onGenerateDocumentId (): void {
+    this.documentId = ''
+    this.isVerifyingDocId = false
+    this.docIdError = null
+  }
+
+  /** Emits an event to confirm the validation state of the Document ID. */
+  @Watch('isValidDocumentId')
+  onValidationUpdate (): void {
+    this.emitIsValidDocumentId(!!this.isValidDocumentId)
+  }
+
+  /**
+   * Emits an event to update the Document ID.
+   * @param docId - The Document ID to emit.
+   */
+  @Emit('updateDocId')
+  emitDocumentId (docId: string): string { return docId }
+
+  /**
+   * Emits an event to update the validation state.
+   * @param isValid - The validation value to emit.
+   */
+  @Emit('isValid')
+  emitIsValidDocumentId (isValid: boolean): boolean { return isValid }
+}
+</script>
+
+<style lang="scss" scoped>
+.title-label {
+  color: #212529FF;
```

Comment:
I think we do have theme colours in this repo. (Also, the hex colours below.)

Also, this particular value looks wrong.

Reply:
Copy that, will double check!

Reply:
Fixed!